File: backend/task-service/src/main/java/io/bento/taskservice/entity/Epic.java
package io.bento.taskservice.entity;

import io.bento.taskservice.enums.EpicStatus;
import lombok.*;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.index.CompoundIndex;
import org.springframework.data.mongodb.core.index.CompoundIndexes;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.Instant;

@Document(collection = "epics")
@CompoundIndexes({
        @CompoundIndex(name = "idx_epic_org_board", def = "{'orgId': 1, 'boardId': 1}"),
        @CompoundIndex(name = "idx_epic_org_id",    def = "{'orgId': 1, 'id': 1}")
})
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Epic {

    @Id
    private String id;

    private String orgId;
    private String boardId;

    private String title;
    private String description;

    /** Hex color used as a label tag, e.g. "#6366f1" */
    private String color;

    @Builder.Default
    private EpicStatus status = EpicStatus.OPEN;

    private Instant startDate;
    private Instant endDate;

    private String ownerId;   // creator / responsible user

    private Instant createdAt;
    private Instant updatedAt;
}

File: frontend/src/types/epic.ts
export type EpicStatus = 'OPEN' | 'IN_PROGRESS' | 'DONE' | 'CANCELLED'

export interface Epic {
  id: string
  orgId: string
  boardId: string
  title: string
  description: string | null
  color: string
  status: EpicStatus
  startDate: string | null
  endDate: string | null
  ownerId: string
  issueCount: number
  createdAt: string
  updatedAt: string
}

export interface CreateEpicData {
  boardId: string
  title: string
  description?: string
  color?: string
  startDate?: string
  endDate?: string
}

export interface UpdateEpicData {
  title?: string
  description?: string
  color?: string
  status?: EpicStatus
  startDate?: string | null
  endDate?: string | null
}
